stream-ingest/src/main/java/com/dellemc/oe/ingest/ImageWriter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package com.dellemc.oe.ingest;
+
+
+import com.dellemc.oe.util.CommonParams;
+import com.dellemc.oe.util.ImageToByteArray;
+import io.pravega.client.ByteStreamClientFactory;
+import io.pravega.client.ClientConfig;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.byteStream.ByteStreamWriter;
+import io.pravega.client.stream.StreamConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.net.URI;
+
+
+/**
+ * A simple example app that uses a Pravega Writer to write to a given scope and stream.
+ */
+public class ImageWriter {
+    // Logger initialization
+    private static final Logger LOG = LoggerFactory.getLogger(ImageWriter.class);
+
+    public final String scope;
+    public final String streamName;
+    public final URI controllerURI;
+
+    public ImageWriter(String scope, String streamName, URI controllerURI) {
+        this.scope = scope;
+        this.streamName = streamName;
+        this.controllerURI = controllerURI;
+    }
+
+    public static void main(String[] args) {
+        final String scope = CommonParams.getScope();
+        final String streamName = CommonParams.getStreamName();
+        final String routingKey = CommonParams.getRoutingKeyAttributeName();
+        final String message = CommonParams.getMessage();
+        final URI controllerURI = CommonParams.getControllerURI();
+        ImageWriter ew = new ImageWriter(scope, streamName, controllerURI);
+        ew.run(routingKey, message);
+    }
+
+    public void run(String routingKey, String message) {
+
+        try {
+            //String scope = "image-scope";
+            String streamName = "image-stream";
+            // Create client config
+            ClientConfig   clientConfig = ClientConfig.builder().controllerURI(controllerURI).build();
+            StreamManager streamManager = StreamManager.create(clientConfig);
+            StreamConfiguration streamConfig = StreamConfiguration.builder().build();
+            if (CommonParams.isPravegaStandalone()) {
+                streamManager.createScope(scope);
+            }
+            streamManager.createStream(scope, streamName, streamConfig);
+
+            // Create ByteStreamClientFactory
+            ByteStreamClientFactory clientFactory = ByteStreamClientFactory.withScope(scope, clientConfig);
+
+            ByteStreamWriter writer = clientFactory.createByteStreamWriter(streamName);
+            //  Read a image and convert to byte[]
+            byte[] payload = ImageToByteArray.readImage();
+
+
+            while (true) {
+                // write image data.
+                writer.write(payload);
```

Comment:
This is not a good example of the byte stream API. There is no way for a reader to separate the stream of bytes into the different images. To do that, you would need to add framing which is very complicated to get right. I recommend removing all examples of the byte stream API.